### ISSUE TITLE
Adding custom maps?

### ISSUE THREAD

**Issue by phugen** [Asunaya/RefinedGunz#21]
Hi, it's me again - another issue that's not really an issue, but rather something I'd like to do:
Is it possible to add custom maps (or rather, maps like High_Haven that were available in late ijji Gunz but aren't currently "shipped" with RGunz)? Simply including the ijji/Aeria-Gunz High_Haven.mrs on the client side and providing a spawn.xml on the server side didn't seem to work, as I'd be greeted by a "Error: Cannot create game" message upon trying to start the map.
I also tried to swap an existing map .mrs for the High_Haven .mrs, bruteforcing its CRC32 value to match the value of the map I swapped out, and changing the spawn.xml so that the spawns are reasonable, but I get the same error. 

Did the map format change after the source leak RGunz is based upon, or why does this happen?

**Comment by Asunaya** [comment]
The RS.xml seems to be encrypted with their MEF stuff. I don't know how to decrypt that (my mrs.exe just yields an empty High_Haven.RS.xml.mef so there's not even anything to decrypt), but since the login map is High Haven apparently, I grabbed the RS.xml from login.mrs ([here](https://pastebin.com/raw/VzhX9txu)), renamed it to High_Haven.RS.xml, and decompressed -> added that file -> recompressed, and it works ok.

The skybox seems to be white though, which I don't think is normal. Not sure what's up with that.

**Comment by phugen** [comment]
Good idea! I thought the login screen was a custom-made login map using pieces of the actual map, so that wouldn't have occured to me at all.

Even though the skybox wouldn't have been a problem considering I want to rip the map from the game as a series of 3D objects, on my end, it works just as intended (and I finally can see what the skybox actually looks like because every version of Gunz I've played so far has this yellow/orange fog effect that makes it basically impossible to see the actual skybox):

http://imgur.com/vIO8FBD

Again, thanks for the quick help. I think this issue could be closed now!

**Comment by Asunaya** [comment]
I think we wanna:

* Add support for loading the MEF stuff natively, so Aeria maps just work (or at the very least add a good error message)
* Fix white skybox
* Add support for the fog stuff in the RS.xml, and maybe an option for disabling it
